### PR TITLE
Upgrade create-starknet to latest version

### PR DESCRIPTION
**Pull request by BDCoy**
Issue #481
This update addresses [Issue #481](https://github.com/apibara/starknet-react/issues/481), which required updating the create-starknet package to use Starknet React v3.

Solution

- Updated the package build system to align with the core package (tsup).
- Updated the templates to create projects that use Starknet React v3, updating the following versions in each project:

"@starknet-react/chains": "^3.0.0-beta.2",
"@starknet-react/core": "^3.0.0-beta.2",
"get-starknet-core": "^3.3.3",
"starknet": "^6.11.0"

- Verified that no breaking changes were introduced.
- Performed a complete test locally on Next.js and Vite projects to ensure compatibility, including testing wallet connections with Argent and Braavos.

### CODE REVIEW

File: packages/create-starknet/public/templates/next/src/components/WalletBar.tsx
Comment (fracek) [inline packages/create-starknet/public/templates/next/src/components/WalletBar.tsx:1]: Be consistent with naming files. Change this to `wallet-bar.tsx`

File: packages/create-starknet/public/templates/next/yarn.lock
Comment (fracek) [inline packages/create-starknet/public/templates/next/yarn.lock:1]: Let's remove the lockfile so users can use what they want.

File: packages/create-starknet/public/templates/vite/src/components/starknet/ConnectModal.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+"use client";
+import { Connector, useConnect } from "@starknet-react/core";
+import React from "react";
+import { Button } from "../ui/Button";
+import Dialog from "../ui/Dialog";
+
+export default function ConnectModal() {
+  const { connect, connectors } = useConnect();
+
+  console.log(connectors);
```
Comment (fracek) [inline packages/create-starknet/public/templates/vite/src/components/starknet/ConnectModal.tsx:10]:
```suggestion
```

File: packages/create-starknet/public/templates/vite/yarn.lock
Comment (fracek) [inline packages/create-starknet/public/templates/vite/yarn.lock:1]: Remove lock file here too.

File: packages/create-starknet/yarn.lock
Comment (fracek) [inline packages/create-starknet/yarn.lock:1]: Remove this lockfile. We use pnpm

**Comment by fracek** [comment]
Looks good, I made some minor comments. You should also add the project to `pnpm-workspace.yaml`

**Comment by BDCoy** [comment]
Hey @fracek,

I was able to update the branch with your comments. I also ran pnpm install-build and updated tsup to version "^8.0.2" to avoid deprecated warnings.

**Comment by fracek** [comment]
Looks like you need to run `pnpm lint:fix`

File: packages/create-starknet/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export { default as nextApp } from "../public/templates/next/src/app/page";
+export { default as viteMain } from "../public/templates/vite/src/main";
```
[the remark this answers is not in the document]
Comment (BDCoy) [reply packages/create-starknet/src/index.ts:2]: Thanks for share the old repo, so basically I just should copy and paste the installation file? I'll run the script

[the remark this answers is not in the document]
Comment (BDCoy) [reply packages/create-starknet/src/index.ts]: I'm done please, run yarn start command to run the package.

**Review by fracek — APPROVED** [review]
Looks good now! Thank you

**Comment by BDCoy** [comment]
Great!

**Pull request merged**